jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -422,24 +463,34 @@ object XGBoost extends Serializable {
       case "_reg_" =>
         val xgBoostModel = new XGBoostRegressionModel(SXGBoost.loadModel(dataInStream))
         setGeneralModelParams(featureCol, labelCol, predictionCol, xgBoostModel)
+      case other =>
+        throw new XGBoostError(s"Unknown model type $other. Supported types " +
+          s"are: ['_reg_', '_cls_'].")
     }
   }
 }
 
+/** Class to hold the training and test matrices in a given partition. */
 private class Watches private(val train: DMatrix, val test: DMatrix) {
-  def toMap: Map[String, DMatrix] = Map("train" -> train, "test" -> test)
-      .filter { case (_, matrix) => matrix.rowNum > 0 }
+
+  def toMap: Map[String, DMatrix] = {
+    Map("train" -> train, "test" -> test).filter { case (_, matrix) => matrix.rowNum > 0 }
+  }
 
   def size: Int = toMap.size
 
-  def delete(): Unit = {
-    toMap.values.foreach(_.delete())
-  }
+  def delete(): Unit = toMap.values.foreach(_.delete())
 
   override def toString: String = toMap.toString
 }
 
 private object Watches {
+
+  /**
+   * Split the data iterator for a Spark partition into train and test `DMatrix`.
```

Comment:
do we really need this? The following three lines are not hard to understand, IMO

```scala
val (trainPoints, testPoints) = labeledPoints.partition(_ => r.nextDouble() <= trainTestRatio)
     val trainMatrix = new DMatrix(trainPoints, cacheFileName)
     val testMatrix = new DMatrix(testPoints, cacheFileName)
```